### DISCUSSION TITLE
Copilot no longer works in debug mode in Visual Studio 2022

### DISCUSSION THREAD

**Discussion by Transigent** [community/community#139288] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I recently updated Visual Studio 2022 to Version 17.11.4 and I have noticed that Copilot no longer gives suggestions when debugging. This is an impediment as I frequently used it in debug mode, in conjunction with Hot Reload to make changes while running. Now I have to stop debug, make changes with Copilot's help, then spend minutes waiting for my program to execute up to the point where I need to debug. 

Has anyone else noticed this? Is there a reason why? I tried to search for suitable terms but didn't find anything similar. Thanks for any advice.

**Comment by hkqr** [comment]
Hello @Transigent 

It seems like you're experiencing an issue with GitHub Copilot not providing suggestions in debug mode after updating to Visual Studio 2022 Version 17.11.4. This is a known issue, and several users have reported similar problems.

From the GitHub Copilot documentation, it's clear that the extension is designed to work in the editor, but it's not explicitly stated that it should work in debug mode. However, many users have reported that it used to work in debug mode before the update.

As a workaround, you can try the following:

Disable and re-enable GitHub Copilot: Sometimes, simply disabling and re-enabling the extension can resolve the issue. To do this, go to Extensions > Manage Extensions, find GitHub Copilot, and toggle the switch to disable it. Then, wait for a few seconds and toggle the switch again to re-enable it.
Reset GitHub Copilot settings: You can try resetting the GitHub Copilot settings to their default values. To do this, go to Tools > Options > GitHub Copilot, and click on the Reset button.
Check for updates: Ensure that you're running the latest version of GitHub Copilot. You can check for updates by going to Extensions > Manage Extensions, finding GitHub Copilot, and clicking on the Update button.
Provide feedback to GitHub: If none of the above workarounds resolve the issue, you can provide feedback to GitHub by going to the GitHub Copilot extension page in the Visual Studio Marketplace and clicking on the Report a problem button.

**Reply by Transigent** [reply]
Thanks for the reply. A lot of your suggestions revolve around Copilot being an extension, which it was before, but if you look at it now it says "Visual Studio now has Github Copilot built in. This extension is now incompatible with your version of Visual Studio." So updating VS as I have done brings the latest version with it. 

I have tried disabling it and reenabling it, I have tried restarting the IDE, restarting the machine, C#, VB.NET etc. etc. and no change. I have updated VS twice in as many weeks (and therefore Copilot) and it has made no difference.

**Reply by Transigent** [reply]
And if I go to https://marketplace.visualstudio.com/items?itemName=GitHub.copilot  and click on the '[Feedback on Copilot completions](https://github.com/orgs/community/discussions/categories/copilot)' link it takes me right back to the main Copilot page where we are now.

**Reply by UserDinDF** [reply]
> Hello @Transigent
> 
> It seems like you're experiencing an issue with GitHub Copilot not providing suggestions in debug mode after updating to Visual Studio 2022 Version 17.11.4. This is a known issue, and several users have reported similar problems.
> 
> From the GitHub Copilot documentation, it's clear that the extension is designed to work in the editor, but it's not explicitly stated that it should work in debug mode. However, many users have reported that it used to work in debug mode before the update.
> 
> As a workaround, you can try the following:
> 
> Disable and re-enable GitHub Copilot: Sometimes, simply disabling and re-enabling the extension can resolve the issue. To do this, go to Extensions > Manage Extensions, find GitHub Copilot, and toggle the switch to disable it. Then, wait for a few seconds and toggle the switch again to re-enable it. Reset GitHub Copilot settings: You can try resetting the GitHub Copilot settings to their default values. To do this, go to Tools > Options > GitHub Copilot, and click on the Reset button. Check for updates: Ensure that you're running the latest version of GitHub Copilot. You can check for updates by going to Extensions > Manage Extensions, finding GitHub Copilot, and clicking on the Update button. Provide feedback to GitHub: If none of the above workarounds resolve the issue, you can provide feedback to GitHub by going to the GitHub Copilot extension page in the Visual Studio Marketplace and clicking on the Report a problem button.

It seems you are not using Copilot, is already integrated in VS and no more add-ons should be installed. After the last update of VS 2022 (17.11.3 and 17.11.4) , Copilot does not work during debug mode, this is related to IntelliSense. I recommend waiting for the update

**Reply by pvw-4bs** [reply]
I'm experiencing the same problem, very frustrating. Can't seem to find an answer anywhere.

**Comment by SimonaLiao** [comment, marked as answer]
Hi,

Thank you for sharing the issue. Previously, we changed the default to turn off completions during debugging based on customer feedback. You can turn it on in Tools > Options > Intellicode > Advanced. Please feel free to share any other thoughts/questions on this issue.

![image](https://github.com/user-attachments/assets/e48f8402-3bd2-4ed1-9c06-9330a1fa9eb9)

**Reply by pvw-4bs** [reply]
Thanks that was indeed the solution. I have to say it was well hidden. I looked through the copilot options, but not Intellicode... As far as I know this wasn't in any release notes? Anyway back to Happy Programming

**Reply by UserDinDF** [reply]
> Hi,
> 
> Thank you for sharing the issue. Previously, we changed the default to turn off completions during debugging based on customer feedback. You can turn it on in Tools > Options > Intellicode > Advanced. Please feel free to share any other thoughts/questions on this issue.
> 
> ![image](https://private-user-images.githubusercontent.com/61256385/371308670-e48f8402-3bd2-4ed1-9c06-9330a1fa9eb9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc0MjIwMDksIm5iZiI6MTcyNzQyMTcwOSwicGF0aCI6Ii82MTI1NjM4NS8zNzEzMDg2NzAtZTQ4Zjg0MDItM2JkMi00ZWQxLTljMDYtOTMzMGExZmE5ZWI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI3VDA3MjE0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkZGUxMDg0NzZiYzVkODQxYTk0YTFiZGJkYjkzYjZhNzFjODA4Y2JjZGQ3ZDQxZTU2OWZjZTNhNDdmNWJhYjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Njgy60xSbyx4217MmhILxJCKE9OL1mF-HLPm0TPbcsI)

thank you it works. It should be more visible because not everyone knows that Copilot while Debug can be enabled in the settings. Oh.................

**Reply by SimonaLiao** [reply]
Thank you for the feedback! We are working on unifying the settings for Intellicode and Copilot in VS to improve that experience, and we will work on increase the discoverability of this option.

**Reply by bastian-weber** [reply]
Oh wow, that is really a weird default. For a while I've been wondering why Copilot would give me so much fewer inline suggestions than my coworkers using VS Code. This solved it. I had looked through the options before, but only in the "Copilot" section of the settings. What's even the difference to IntelliCode? Maybe these two should be unified?

**Reply by SimonaLiao** [reply]
Great point! We are working on the unification now :) To provide more info on IntelliCode, it also provides code completions using a locally running LLM (which is smaller than gpt models) but it's only for C# in VS. [ You can read more about it here](https://visualstudio.microsoft.com/services/intellicode/).

**Reply by 6EyesStudioLLC** [reply]
Secretly disabling features (such as all of copilot) as the default behavior is certainly not the right way. Adding the option is good, but the option should have started 'turned off', so that people that are looking for a change of behavior can change the feature, and those that aren't aware of the issue don't have copilot turned off without their knowledge. Yes, even if there was "customer feedback".

**Reply by GTechMicah** [reply]
Thank You so much, finally found this because I've been wondering why it didn't work during debugging for a while now, Thank You!

**Reply by rbuss93** [reply]
Thanks, this worked for me! Reinstalled VS and it didn't work while debugging. After removing the checkmark, it's working again. "False" should be the default.

**Reply by bastian-weber** [reply]
Is this setting still there in Visual Studio 2026, or has it been removed?

**Reply by rbuss93** [reply]
> Is this setting still there in Visual Studio 2026, or has it been removed?

It has been moved in VS2026:
<img width="1063" height="967" alt="image" src="https://github.com/user-attachments/assets/69e36f91-0a4a-4cdc-a304-87bab1d7f029" />

**Reply by bastian-weber** [reply]
Thanks a lot for pointing that out! 

Apparently it had been turned off again even though I had imported my settings from VS 2022. I can't stress this enough, it's  a terrible default.
